app/lib/util/search/search-service.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:inside_data_flutter/inside_data_flutter.dart';
+import 'package:rxdart/rxdart.dart';
+
+//TODO: Overkill?
```

Comment:
Most things in this file can be moved to a more general place.
The only thing left is keeping track of current query value.
There's probably a better way to keep track of state? But that's fine, we don't have to fix every little thing.
(See, I compromise sometimes)